conjure/src/main/java/com/palantir/conjure/cli/ConjureCli.java

```diff
@@ -58,61 +57,80 @@ public static void main(String[] args) {
         System.exit(prepareCommand().execute(args));
     }
 
+    public static void inProcessExecution(String[] args) {
+        ExceptionHandler exceptionHandler = new ExceptionHandler();
+        new CommandLine(new ConjureCli())
+                .setExecutionExceptionHandler(exceptionHandler)
+                .execute(args);
+        exceptionHandler.maybeRethrowException();
+    }
+
     @VisibleForTesting
     static CommandLine prepareCommand() {
         return new CommandLine(new ConjureCli()).setExecutionExceptionHandler(new ExceptionHandler());
     }
 
     public static final class ExceptionHandler implements IExecutionExceptionHandler {
+
+        private Optional<Exception> thrownException = Optional.empty();
```

Comment:
I actually think we can write this a little more concisely/safely for future changes. We can keep the existing `IExecutionExceptionHandler` as is then write a wrapper `IExecutionExceptionHandler` that wraps the original one but catches the exception and saves it. Also sets the `err` and if that is written to, we make an exception from that.

Reply:
Perhaps with a `TeeOutputStream` to the original `err` if we want to preserve the current behaviour (maybe we don't need to though?)

Reply:
added a separate class that does the exception wrapping here: https://github.com/palantir/conjure/pull/1516/files#diff-cc1e8eaeed51e87c5b283c659588f4c2f976ceafdc8ad587e89fc92460fdccedR33.

Reply:
Nice!